avocado/utils/archive.py

```diff
@@ -59,23 +59,46 @@ def gzip_uncompress(path, output_path):
         return output_path
 
 
-class ArchiveException(Exception):
+def is_lzma_file(path):
     """
-    Base exception for all archive errors.
+    Checks if file given by path has contents that suggests lzma file
     """
+    with lzma.LZMAFile(path, 'rb') as lzma_file:
+        try:
+            _ = lzma_file.read(1)
+        except lzma.LZMAError:
+            return False
+    return True
 
 
-def extract_lzma(path, force=False):
+def extract_lzma(path, output_path=None, force=False):
     """
     Extracts a XZ compressed file to the same directory.
     """
-    extracted_file = os.path.splitext(path)[0]
-    if not force and os.path.exists(extracted_file):
-        return extracted_file
+    if output_path is None:
+        output_path = os.path.splitext(path)[0]
+    elif os.path.isdir(output_path):
+        basename = os.path.basename(path)
+        if basename.endswith('.xz'):
+            basename = os.path.splitext(basename)[0]
+        output_path = os.path.join(output_path, basename)
+    if not force and os.path.exists(output_path):
+        return output_path
     with lzma.open(path, 'rb') as file_obj:
-        with open(extracted_file, 'wb') as newfile_obj:
+        with open(output_path, 'wb') as newfile_obj:
             newfile_obj.write(file_obj.read())
-    return extracted_file
+    return output_path
+
+
+# TODO: remove extract_lzma in favor of lzma_uncompress to make
```

Comment:
I think it could be done in this PR. I was looking that in avocado there is just one usage of `extract_lzma` so another PR is not necessary.
https://github.com/avocado-framework/avocado/blob/master/avocado/utils/vmimage.py#L351

Reply:
OK, no objections here.